src/utility/utility.cpp

```diff
@@ -115,4 +116,60 @@ double EllipIntSecond(const double &x)
   return result;
 }
 
+bool almost_the_same(const double &a,
+                     const double &b,
+                     const double &rel_precision,
+                     const double &num_zero)
+{
+  if (std::abs(a) < num_zero and std::abs(b) < num_zero)
+  {
+    return true;
+  }
+  return std::abs(a - b) < std::abs(a + b) / 2 * rel_precision;
+}
+
+bool almost_the_same(const std::complex<double> &a,
```

Comment:
This function is not tested by tests.

Reply:
This function should now be tested for in `Test-gw.cpp`.